xwiki-platform-core/xwiki-platform-attachment/xwiki-platform-attachment-api/src/main/java/org/xwiki/attachment/internal/job/MoveAttachmentJob.java

```diff
@@ -0,0 +1,157 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.attachment.internal.job;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.concurrent.ThreadLocalRandom;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+
+import org.xwiki.attachment.MoveAttachmentRequest;
+import org.xwiki.attachment.internal.AttachmentsManager;
+import org.xwiki.attachment.internal.RedirectAttachmentClassDocumentInitializer;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.localization.ContextualLocalizationManager;
+import org.xwiki.model.reference.AttachmentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+import org.xwiki.refactoring.internal.job.AbstractEntityJob;
+import org.xwiki.refactoring.job.EntityJobStatus;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiAttachment;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+import static org.apache.commons.lang.exception.ExceptionUtils.getRootCauseMessage;
+import static org.xwiki.attachment.internal.RedirectAttachmentClassDocumentInitializer.SOURCE_NAME_FIELD;
+import static org.xwiki.attachment.internal.RedirectAttachmentClassDocumentInitializer.TARGET_LOCATION_FIELD;
+import static org.xwiki.attachment.internal.RedirectAttachmentClassDocumentInitializer.TARGET_NAME_FIELD;
+
+/**
+ * This is the job in charge of moving attachments.
+ *
+ * @version $Id$
+ * @since 14.0RC1
+ */
+@Component
+@Named(MoveAttachmentJob.HINT)
+public class MoveAttachmentJob
+    extends AbstractEntityJob<MoveAttachmentRequest, EntityJobStatus<MoveAttachmentRequest>>
+{
+    /**
+     * The hint for this job.
+     */
+    public static final String HINT = "refactoring/attachment/move";
+
+    @Inject
+    private Provider<XWikiContext> xcontextProvider;
+
+    @Inject
+    private EntityReferenceSerializer<String> entityReferenceSerializer;
+
+    @Inject
+    private ContextualLocalizationManager contextualLocalizationManager;
+
+    @Inject
+    private EntityReferenceSerializer<String> referenceSerializer;
+
+    @Inject
+    private AttachmentsManager attachmentsManager;
+
+    @Override
+    public String getType()
+    {
+        return HINT;
+    }
+
+    @Override
+    protected void process(EntityReference source)
+    {
+        this.progressManager.pushLevelProgress(this);
+        this.request.setId("refactoring", "moveAttachment",
```

Comment:
hmmm wondering how much it would be useful to reuse the source in the id? Not sure if it's a practice we do for other refactorings and/or if it might help in case of problem to find back the proper logs / info.

Reply:
@tmortagne wdyt? Could be interesting but that's not what we did for the document move for instance.

Reply:
First, I don't understand why you set the id during job execution, that does not really make any sense. The id is supposed to be set when you create the request before executing the job.

> that's not what we did for the document move for instance.

As for having the source in the id, the other refactoring jobs are using the source for the group id but indeed not for the request id from what I can see. It could indeed be interesting for debug to have it, but it means potentially having a path that simply does not work on various systems like Windows because it's too long.

Reply:
> it means potentially having a path that simply does not work on various systems like Windows because it's too long.

ok I forgot that aspect. So not such a good idea.